rpcs3/Emu/RSX/Common/TextureUtils.cpp

```diff
@@ -53,6 +53,38 @@ struct copy_unmodified_block_swizzled
 	}
 };
 
+struct copy_unmodified_block_vtc
+{
+	template<typename T, typename U>
+	static void copy_mipmap_level(gsl::span<T> dst, gsl::span<const U> src, u16 width_in_block, u16 row_count, u16 depth, u32 dst_pitch_in_block, u32 src_pitch_in_block)
+	{
+
+		static_assert(sizeof(T) == sizeof(U), "Type size doesn't match.");
+		u32 row_element_count = width_in_block * row_count;
+		u32 dst_offset = 0;
+		u32 src_offset = 0;
+		const u32 vtc_tile_row_count = 4;
+
+		// Undo Nvidia VTC tiling - place each 2D texture slice back to back in linear memory
+		for (int d = 0; d < depth; d++) {
+
+			// copy one slice of the 3d texture
+			for (u32 i = 0; i < row_element_count; i += 1) {
+				copy(dst.subspan(dst_offset + i, 1), src.subspan(src_offset + i * vtc_tile_row_count, 1));
+			}
+
+			dst_offset += row_element_count;
+
+			if (d && ((d & (vtc_tile_row_count - 1)) == 0)) {
```

Comment:
brace on newline;

also the test here is a little confusing; skipping one slice every 4 other slices if I understand this correctly (4, 8, 12, 16, etc). Clarify why this is needed if it is intentional.
Also, the spec specifies special handling when `(d > LARGEST_POWER_OF_2_LESS_THAN_DEPTH)`, does this algorithm handle this correctly?

Reply:
I'll try to put some more documentation in the code.  I agree the memory layout is confusing/complex.  

I'll investigate the power of 2 depth.